compiler/daml-lf-proto/src/DA/Daml/LF/Mangling.hs

```diff
@@ -108,39 +109,42 @@ mangleIdentifier txt = case T.foldl' f (MangledSize 0 0) txt of
 
 unmangleIdentifier :: T.Text -> Either String T.Text
 unmangleIdentifier txt = do
-  chs <- goStart (T.unpack txt)
-  if null chs
+  chs <- goStart txt
+  if T.null chs
     then Left "Empty identifier"
-    else Right (T.pack chs)
+    else Right chs
   where
-    go :: (Char -> Bool) -> (String -> Either String String) -> String -> Either String String
-    go isAllowed followUp = \case
-      [] -> Right []
-      ['$'] -> Left "Got control character $ with nothing after it. It should be followed by $, u, or U"
-      '$' : ctrlCh : chs0 -> case ctrlCh of
-        '$' -> ('$' :) <$> followUp chs0
-        'u' -> do
-          (ch, chs1) <- readEscaped "$u" 4 chs0
-          (ch :) <$> followUp chs1
-        'U' -> do
-          (ch, chs1) <- readEscaped "$U" 8 chs0
-          (ch :) <$> followUp chs1
-        ch -> Left ("Control character $ should be followed by $, u, or U, but got " ++ show ch)
-      ch : chs -> if isAllowed ch
-        then (ch :) <$> followUp chs
-        else Left ("Unexpected unescaped character " ++ show ch)
-    
+    go :: (Char -> Bool) -> (T.Text -> Either String T.Text) -> T.Text -> Either String T.Text
+    go isAllowed followUp s
```

Comment:
If I'm not missing something, `followUp` is instantiated with `goPart` on all call sites.

Reply:
`followUp` is dead.